File: PremiereNumber/app/src/main/java/com/example/premierenumber/MainActivity.java
package com.example.premierenumber;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    protected TextView txt;
    protected EditText input;
    protected ProgressBar bar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        txt = findViewById(R.id.txt);
        input = findViewById(R.id.editText);
        bar = findViewById(R.id.progress);

        bar.setMax(100);
        bar.setProgress(0);
    }

    public void asyncCheckNum(View view) {
        txt.setVisibility(View.VISIBLE);
        txt.setText(" ");
        bar.setVisibility(View.VISIBLE);
        MonAsyncTask asyncTask = new MonAsyncTask(this);
        asyncTask.execute(Long.parseLong(input.getText().toString()));
    }

    @SuppressLint("SetTextI18n")
    public void getRes(Boolean resultat){
        if (resultat){
            txt.setText("Le nombre est premier!");
            txt.setTextColor(Color.GREEN);
        } else {
            txt.setText("Le nombre n'est pas premier!");
            txt.setTextColor(Color.RED);
        }
    }
}

File: PremierNumberKotlin/app/src/main/java/com/formationandroid/premiernumberkotlin/MainActivity.kt
package com.formationandroid.premiernumberkotlin

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers.Default
import kotlinx.coroutines.Dispatchers.Main
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import kotlin.math.ceil
import kotlin.math.sqrt

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    @SuppressLint("SetTextI18n")
    fun clickIsPremier(view: View){
        CoroutineScope(Default).launch {
            if (editText.text.toString().isNotEmpty()){
                val nbToDeterminate: Long = editText.text!!.toString().toLong()
                val retour = getIfPremier(nbToDeterminate)
                withContext(Main) {
                    resultat.visibility = View.VISIBLE
                    if (retour) {
                        resultat.setTextColor(Color.RED)
                        resultat.text = "$nbToDeterminate n'est pas premier!"
                    } else {
                        resultat.setTextColor(Color.GREEN)
                        resultat.text = "$nbToDeterminate est un nombre premier!"
                    }
                    editText.text.clear()
                    progressBar.visibility = View.INVISIBLE
                }
            }
        }
    }

    private suspend fun getIfPremier(nbToDeterminate: Long): Boolean {
        var flag = false
        withContext(Main) {
            progressBar.visibility = View.VISIBLE
        }
        // condition for nonprime number
        for (i: Long in 2L..ceil(sqrt(nbToDeterminate.toDouble())).toLong()) {
            if (i%10000L == 0L) {
                withContext(Main) {
                    progressBar.progress = ((i * 100) / nbToDeterminate).toInt()
                }
            }
            if (nbToDeterminate % i == 0L) {
                flag = true
                break
            }
        }

        return flag
    }
}
